lib/ex_unit/test/ex_unit/assertions_test.exs

```diff
@@ -127,6 +127,22 @@ defmodule ExUnit.AssertionsTest do
     end
   end
 
+  test "assert received with multiple pinned variable" do
+    status = :valid
+    send self(), {:status, :invalid, :invalid}
+    try do
+      "This should never be tested" = assert_received {:status, ^status, ^status}
+    rescue
+      error in [ExUnit.AssertionError] ->
+        "No message matching {:status, ^status, ^status} after 0ms.\n" <>
+        "The following variables were pinned:\n" <>
+        "  status = :valid\n" <>
+        "  status = :valid\n" <>
```

Comment:
We should likely not show status twice. :D Let's call `Enum.uniq` on the names but still leave this test here (showing that dup variables will be showed once).